compiler/src/main/kotlin/se/ansman/kotshi/generators/DataClassAdapterGenerator.kt

```diff
@@ -158,18 +162,32 @@ class DataClassAdapterGenerator(
             .addParameter(writerParameter)
             .addParameter(value)
 
+        val label = element.getAnnotation(PolymorphicLabel::class.java)?.value
+        val labelKey = types.asElement(element.superclass)?.getAnnotation(Polymorphic::class.java)?.labelKey
+
         if (properties.isEmpty()) {
```

Comment:
I think this whole if-statement is a dead now since we started parsing Kotlin metadata. Kotlin does not allow data classes with an empty constructor so go ahead and remove the `if` part.